components/molecule/phoneInput/src/index.js

```diff
@@ -82,15 +85,27 @@ export default function MoleculePhoneInput({
           </AtomIcon>
         </div>
         <div className={`${baseClass}-input-phoneContainer`}>
-          <p className={`${baseClass}-input-prefix-code`}>
-            {selectedPrefix.countryCode}
-          </p>
+          {selectedPrefix && (
+            <p className={`${baseClass}-input-prefix-code`}>
+              {selectedPrefix.countryCode}
+            </p>
+          )}
           <AtomInput
             value={value.toString()}
-            mask={selectedPrefix.mask || '000 000 000'}
+            mask={
+              selectedPrefix && {
+                mask: isLandLine
+                  ? selectedPrefix.mask.landlineMask
+                  : selectedPrefix.mask.mobileMask
+              }
+            }
             placeholder={placeholder}
             type={TYPES.MASK}
             onChange={(e, {value}) => {
+              selectedPrefix.landlinePrefixs.includes(value[0])
```

Comment:
I'm fan of `ternary operator` but sometimes decrease the readability of a file. Maybe here is not the best scenario to use it. What do you think?

Reply:
i'll do a full refactor of the function and maybe it will end up being more readable given the ideas you've provided before!